### PR TITLE
fix: keys setting vulns

### PR DESCRIPTION
**Pull request by ogzhanolguncu**
## What does this PR do?

Fixes vulnerabilities in key settings, prevents the exposure of workspaceId to RPC.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- The `DefaultBytes` and `DefaultPrefix` components have been updated to remove the `workspaceId` field, simplifying the form structure.

- **Bug Fixes**
	- Enhanced error handling for the `setDefaultBytes` and `setDefaultPrefix` procedures by streamlining the input validation process.

- **Refactor**
	- Adjusted the input schemas and props in both components to eliminate dependencies on `workspaceId`, improving overall functionality.

### CODE REVIEW

**Review by perkinsjr — APPROVED** [review]

**Review by chronark — CHANGES_REQUESTED** [review]
this is not secure

File: apps/dashboard/lib/trpc/routers/api/setDefaultBytes.ts [hunk 1]
```diff
@@ -30,7 +29,7 @@ export const setDefaultApiBytes = t.procedure
             "We were unable to find the KeyAuth. Please try again or contact support@unkey.dev.",
         });
       });
-    if (!keyAuth || keyAuth.workspaceId !== input.workspaceId) {
```
Comment (chronark) [inline apps/dashboard/lib/trpc/routers/api/setDefaultBytes.ts:33]: why do we make it less secure?

we should never ever ever read a workspaceId from input, only ever from context

File: apps/dashboard/lib/trpc/routers/api/setDefaultPrefix.ts [hunk 2]
```diff
@@ -25,7 +24,7 @@ export const setDefaultApiPrefix = t.procedure
           message: "We were unable to find KeyAuth. Please try again or contact support@unkey.dev.",
         });
       });
-    if (!keyAuth || keyAuth.workspaceId !== input.workspaceId) {
```
Comment (chronark) [inline apps/dashboard/lib/trpc/routers/api/setDefaultPrefix.ts:28]: same as above

**Review by chronark — CHANGES_REQUESTED** [review]

File: apps/dashboard/lib/trpc/routers/api/setDefaultBytes.ts (hunk 1 above)
Comment (ogzhanolguncu) [reply apps/dashboard/lib/trpc/routers/api/setDefaultBytes.ts:33]: Am I missing something? Previously we were reading from input now its omitted and just checking if keyAuth exists?

**Review by chronark — APPROVED** [review]

**Pull request merged**